src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,272 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.exception.DivisionByZeroException;
+import core.basesyntax.exception.NotSupportOperationException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
+    private static final char ADDITION = '+';
+    private static final char SUBTRACTION = '-';
+    private static final char DIVISION = '/';
+    private static final char MULTIPLICATION = '*';
+    private static final char TO_POWER = '^';
+    private static final double DELTA = 0.0001;
+    private Calculator calculator;
+    private double firstPositive;
+    private double secondPositive;
+    private double firstNegative;
+    private double secondNegative;
+    private double zero;
+
+    @BeforeEach
+    void setUp() {
+        calculator = new Calculator();
+        firstPositive = 16.234;
+        firstNegative = -16.234;
+        secondPositive = 15.234;
+        secondNegative = -15.234;
+        zero = 0.0;
+    }
+
+    @Test
+    void twoPositiveAddition_OK() {
+        double expected = 31.468;
+        assertEquals(expected, calculator.calculate(firstPositive, secondPositive,
+                ADDITION), DELTA);
```

Comment:
I am not sure about those result in expected, it could be like 
```suggestion
        double expected = firstPositive + secondPositive;
        assertEquals(expected, calculator.calculate(firstPositive, secondPositive,
                ADDITION), DELTA);
```